pkg/client/rootd/dns/server.go

```diff
@@ -213,6 +216,19 @@ func (s *Server) shouldDoClusterLookup(query string) bool {
 	return true
 }
 
+func (s *Server) isExcluded(query string) bool {
+	s.configLock.RLock()
+	defer s.configLock.RUnlock()
+	// When intercepting, this function will potentially receive the hostname of any search param, so their
+	// unqualified hostname should be evaluated too.
+	unqualifiedHostnames := make([]string, 0)
+	for i := range s.search {
+		withoutDomain := strings.TrimSuffix(query, s.search[i])
+		unqualifiedHostnames = append(unqualifiedHostnames, strings.TrimSuffix(withoutDomain, "."))
+	}
+	return slice.ContainsAny(s.config.Excludes, append(unqualifiedHostnames, query))
```

Comment:
This is a slow way of doing this, involving building that contains elements that perhaps don't have a matching suffix, which is then scanned sequentially multiple times, only to return a boolean if there's a hit. Also, the last line append `query` without stripping its trailing dot. 

How about something like this?
```suggestion
	if slice.Contains(s.config.Excludes, query[:len(query)-1]) {
		return true
	}
	for _, sp := range s.search {
		if strings.HasSuffix(query, sp) && slice.Contains(s.config.Excludes, query[:len(query)-len(sp)-1]) {
			return true
		}
	}
	return false
```